### PR TITLE
More complete CI

### PR DESCRIPTION
**Pull request by maxulysse**
Following @Emiller88 on https://github.com/nf-core/rnaseq/pull/883

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Launching all tests with singularity and conda and will add exclude in the GHA workflow to remove the ones that don't work (thinking mainly about conda + annotation)

File: .github/workflows/pytest-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+          - profile: "conda"
+            tags: deepvariant
+          - profile: "conda"
+            tags: haplotypecaller
+          - profile: "conda"
+            tags: merge
+          - profile: "conda"
+            tags: snpeff
+          - profile: "conda"
+            tags: vep
```
Comment (Emiller88) [inline .github/workflows/pytest-workflow.yml:51]: Clever!

File: tests/test_cnvkit.yml [hunk 2]
```diff
@@ -0,0 +1,158 @@
+- name: Run variant calling on somatic samples with cnvkit
```
Comment (Emiller88) [inline tests/test_cnvkit.yml:1]: Might consider splitting these into their own individual files `tests/workflows/cnvkit/test_variantcalling_somatic.yml` or something for readability, these are going to get quite long.

**Review by Emiller88 — APPROVED** [review]
Awesome work and a massive effort!

Only thought is you might consider splitting it into a `workflows` directory or something to split it apart from `subworkflows` and `modules` for other local tests in the future, but that can be addressed when we get there.

File: tests/test_cnvkit.yml (hunk 2 above)
Comment (maxulysse) [reply tests/test_cnvkit.yml:1]: I don't think there will be much more tests for cnvkit actually, but I'll think about that.
Thanks

**Comment by maxulysse** [comment]
> Awesome work and a massive effort!
> 
> Only thought is you might consider splitting it into a `workflows` directory or something to split it apart from `subworkflows` and `modules` for other local tests in the future, but that can be addressed when we get there.
> 
> Also, you might consider not running singularity and conda on every commit to a PR as well, there are a lot of tests here. So not only taking up a ton of runners and energy, but you might hit the matrix limit.

Was thinking more not to run the `modules` or `subworkflows` tests as I'm assuming we'll run that in `nf-core/modules`.
Maybe just for the local ones, but will need to figure this out....

**Comment by Emiller88** [comment]
BTW, sorry for figuring this out just as you set this up :see_no_evil: https://github.com/nf-core/methylseq/blob/dev/.github/workflows/ci.yml

**Review by FriederikeHanssen — APPROVED** [review]
Love the splitting of the modules.config

**Pull request merged**